### DISCUSSION TITLE
Remapping UART to I2C in Beast F7 45A V1.2 AIO

### DISCUSSION THREAD

**Discussion by cjmv** [iNavFlight/inav#7416] [Q&A]
This is probably a dumb question.

I have the impression (from past reading, but I can't find a reference) that it wasn't possible to remap resources in INAV. Not sure if that also/still applies for UARTs. In this particular case the Beast F7 45A V1.2 AIO doesn't have a I2C pads broken out in the board, so I was wondering if it is possible to remap a UART to I2C through CLI, or if I would have to build my own custom firmware to achieve that.

Of course I could also buy an AIO FC that has it already broken out but that's my bad... I forgot to check for it.

**Comment by macleayislandbruce** [comment]
I am fairly sure this would have to be custom firmware. It should be possible to remap an i2c bus to other pins in firmware but I am not familiar with this board.

**Comment by macleayislandbruce** [comment]
I have had a look at the iFlight Beast, it uses STM32F745 the same as the OMNIBUSF7V2 I am working on. I am doing something similar in that I have a conflict with UART3 and i2c sharing pins. I cannot see any iNav or Betaflight targets for your board, is there any source code for your board?

**Reply by zgliyun** [reply]
I contact iflight tech support and get the anwser:
use CLI to export I2C interface: (this is Betaflight command CLI) 
1. Connect eCompass SCL to T3, SDA to R3.
2. resource I2C_SCL 2 B10
3. resource I2C_SDA 2 B11
4. set mag_i2c_device = 2
5. save

I enabled eCompass for BF, and going to try INAV this time.

**Reply by zgliyun** [reply]
remenber to disable UART3 in settings.

**Reply by reednoel4u** [reply]
Dear
My FC is beast Alo v1.2, 
I not see T3 pad, R3 is in DJI port!
Please show me where is that

**Comment by BryanFPV** [comment]
same question here, I was trying to flash my Beast F7 45A AIO with INAV firmware, but i failed to find a target for this FC. I'm curious which target were you used for your Beast F7 FC?

**Reply by Ujjwaana** [reply]
Did iNav got Firmware support for iFlight Beast ? if not , then it is such a shame. iFlight is not an unknown brand and people not able to use it for iNav is such a handicap.

**Reply by andersoncalixto31** [reply]
Eu comprei uma placa iFlight Best F745 AIO e não encontrei firmware no INAV, posso usar outra no lugar?

**Comment by zgliyun** [comment]
I've made the customized FW for IFLIGHT_F745_AIOV2.
If anybody want the code, just send me an E-mail: zgliyun@gmail.com

**Reply by G2Giaco** [reply]
@zgliyun thanks so much for the code - however the accelerometer is not working when I load it into iNav though.  any thoughs?  Can you share your dump file?

**Reply by reednoel4u** [reply]
> I've made the customized FW for IFLIGHT_F745_AIOV2. If anybody want the code, just send me an E-mail: [zgliyun@gmail.com](mailto:zgliyun@gmail.com)

my fc is Beast F745 v1.2, could you please help me fw of inav? I already send email to you.
Thanks

**Comment by andersoncalixto31** [comment]
Bom dia,Obrigado por responder minha pergunta e agradeço, fico no aguardo do envio.Abs,AndersonEm 4 de mai. de 2022 03:50, zgliyun ***@***.***> escreveu:
I've made the customized FW for IFLIGHT_F745_AIOV2.
If anybody want the code, just send me an E-mail: ***@***.***
